test/services/MetricsMonitoringServiceSpec.scala

```diff
@@ -17,49 +17,57 @@
 package services
 
 import com.codahale.metrics.{Meter, MetricRegistry}
-import model.{AuditContext, Location, TAuditContext}
+import engine.{AuditInfo, ThrottlingInfo}
+import model.Location
 import org.mockito.Matchers.{any, eq => eqTo}
 import org.mockito.Mockito._
 import org.scalatest.concurrent.Eventually
 import org.scalatest.mock.MockitoSugar
 import org.scalatest.prop.TableDrivenPropertyChecks._
+import org.scalatest.prop.TableFor3
 import org.scalatest.prop.Tables.Table
 import play.api.test.FakeRequest
 import uk.gov.hmrc.play.http.HeaderCarrier
 import uk.gov.hmrc.play.test.UnitSpec
 
-import scala.collection.mutable.{Map => mutableMap}
-
 class MetricsMonitoringServiceSpec extends UnitSpec with MockitoSugar with Eventually {
 
   "MetricsMonitoringService" should {
 
-    val scenarios = Table(
+    val scenarios: TableFor3[String, Option[Location], Location] = Table(
       ("scenario", "destinationNameBeforeThrottling", "destinationNameAfterThrottling"),
-      ("destination not throttled", Some("some-destination"), "some-destination"),
-      ("destination throttled", Some("some-destination"), "some-other-destination"),
-      ("destination before throttling not present", None, "some-other-destination")
+      ("destination not throttled", Some(Location(name = "a", url = "/a")), Location(name = "a", url = "/a")),
+      ("destination throttled", Some(Location(name = "a", url = "/a")), Location(name = "b", url = "/b")),
+      ("destination before throttling not present", None, Location(name = "b", url = "/b"))
     )
 
-    forAll(scenarios) { (scenario: String, destinationNameBeforeThrottling: Option[String], destinationNameAfterThrottling: String) =>
+    forAll(scenarios) { (scenario, destinationNameBeforeThrottling, destinationNameAfterThrottling) =>
 
       s"send monitoring events - scenario: $scenario" in new Setup {
 
-        val mockAuditContext = mock[TAuditContext]
+        import engine.RoutingReason._
 
-        val mockThrottledLocation = mock[Location]
-        when(mockThrottledLocation.name).thenReturn(destinationNameAfterThrottling)
+        // given
+        val c1 = IS_A_VERIFY_USER
+        val c2 = IS_A_GOVERNMENT_GATEWAY_USER
+        val c3 = IS_IN_A_PARTNERSHIP
+        val c4 = IS_SELF_EMPLOYED
```

Comment:
can these conditions be named something more expressive? and also can they be moved to `Setup` (along with the 'meter' mocking below) ?